addons/web_editor/static/lib/odoo-editor/src/commands/deleteBackward.js

```diff
@@ -164,13 +164,15 @@ HTMLElement.prototype.oDeleteBackward = function (offset, alreadyMoved = false,
          */
         if (
             !this.previousElementSibling &&
-            ['BLOCKQUOTE', 'H1', 'H2', 'H3', 'PRE'].includes(this.nodeName) &&
+            ['BLOCKQUOTE', 'H1', 'H2', 'H3', 'H4', 'H5', 'H6', 'PRE'].includes(this.nodeName) &&
```

Comment:
This is way better, thanks! Kind of nitpicking but at this point I'd use `paragraphRelatedElements` and just add a check for `this.nodeName !== 'P'` rather than relisting everything like that.